simbad2k/simbad2k.py

```diff
@@ -152,12 +152,16 @@ def get_primary_designation(self):
                 if self.scheme_mapping[self.scheme] == 'asteroid':
                     try:
                         return int(target['permid']), None
-                    except (ValueError, KeyError):
+                    except (ValueError, KeyError, TypeError):
                         continue
                 elif self.scheme_mapping[self.scheme] == 'comet':
                     perm_id = target.get('permid')
                     if perm_id:
-                        return perm_id, None
+                        try:
+                            int(target['permid'])
```

Comment:
What does this do?

Reply:
I've added some comments. Basically it's a manual check that the type of object we found matches the type of object we were looking for.